crates/type/src/user/record.rs

```diff
@@ -1,12 +1,80 @@
 // SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
 // SPDX-License-Identifier: AGPL-3.0-only
 
-use serde::{Deserialize, Serialize};
+use std::fmt;
+
+use serde::{
+    de::{self, Deserializer, MapAccess, Visitor},
+    Deserialize, Serialize,
+};
 
 /// A user record
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct Record {
-    /// OpenID Connect identity subject uniquely identifying the user
-    pub subject: String,
+    /// OpenID Connect identity subjects uniquely identifying the users with access
+    pub subjects: Vec<String>,
+}
+
+impl Record {
+    pub fn new(subject: String) -> Self {
+        Self {
+            subjects: vec![subject],
+        }
+    }
```

Comment:
If we need a `new` here, it should take an `impl IntoIterator<Item=String>` as an argument, not a singular `String`